From a3129f3d2cfc09bddb7ccb0baa54bdc4c8528018 Mon Sep 17 00:00:00 2001
From: Hao Zheng <haoz@marvell.com>
Date: Tue, 10 Sep 2019 22:13:42 -0700
Subject: [PATCH 341/767] octeontx2-af: kpu profile fix for a missing action
 entry

commit f03662a20e69b916ae13ac28841381962814b358 from
git@git.assembla.com:cavium/WindRiver.linux.git

This patch fixes a problem where the catchall entry in KPU2 misses
the corresponding action entry.

Change-Id: I5525b5029623007af087170ca30e24592fcf8e6e
Signed-off-by: Hao Zheng <haoz@marvell.com>
Reviewed-on: https://sj1git1.cavium.com/15602
Reviewed-by: Sunil Kovvuri Goutham <Sunil.Goutham@cavium.com>
Tested-by: Sunil Kovvuri Goutham <Sunil.Goutham@cavium.com>
Signed-off-by: Kevin Hao <kexin.hao@windriver.com>
---
 drivers/net/ethernet/marvell/octeontx2/af/npc_profile.h | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/drivers/net/ethernet/marvell/octeontx2/af/npc_profile.h b/drivers/net/ethernet/marvell/octeontx2/af/npc_profile.h
index 9cb7f68b5b9f..0f52e2df4698 100644
--- a/drivers/net/ethernet/marvell/octeontx2/af/npc_profile.h
+++ b/drivers/net/ethernet/marvell/octeontx2/af/npc_profile.h
@@ -8385,6 +8385,14 @@ static struct npc_kpu_profile_action kpu2_action_entries[] = {
 		NPC_F_LB_U_UNK_ETYPE | NPC_F_LB_L_EXDSA,
 		0, 0, 0, 0,
 	},
+	{
+		NPC_ERRLEV_LB, NPC_EC_L2_K3,
+		0, 0, 0, 0, 1,
+		NPC_S_NA, 0, 0,
+		NPC_LID_LB, NPC_LT_NA,
+		0,
+		0, 0, 0, 0,
+	},
 };
 
 static struct npc_kpu_profile_action kpu3_action_entries[] = {
-- 
2.31.1

